app/cdap/components/hydrator/components/SidePanel/SidePanel.tsx

```diff
@@ -153,14 +286,19 @@ export const SidePanel = ({
         // treat plugin group plugins as immutable
         const newPluginGroups = pluginGroups.map((group) => {
           const newGroup = { ...group, plugins: [] };
-          newGroup.plugins = group.plugins.map((plugin) => {
-            const newPlugin = { ...plugin };
-            // add the display name and show custom icon to the plugins
-            newPlugin.displayName = generateLabel(plugin, all.plugins.pluginsMap);
-            newPlugin.showCustomIcon = shouldShowCustomIcon(plugin, all.plugins.pluginsMap);
-            newPlugin.customIconSrc = getCustomIconSrc(plugin, all.plugins.pluginsMap);
-            return newPlugin;
-          });
+          newGroup.plugins = group.plugins
+            .map((plugin) => {
+              const newPlugin = { ...plugin };
+              // add the display name and show custom icon to the plugins
+              newPlugin.displayName =
+                generateLabel(plugin, all.plugins.pluginsMap) || newPlugin.name;
+              newPlugin.showCustomIcon = shouldShowCustomIcon(plugin, all.plugins.pluginsMap);
+              newPlugin.customIconSrc = getCustomIconSrc(plugin, all.plugins.pluginsMap);
+              return newPlugin;
```

Comment:
Nit: Could do
```
const newPlugin = {
  ...plugin,
  displayName: foo,
  showCustomIcon: bar,
  customIconSrc: baz
};
```

or even just return that without creating a variable.

Reply:
yeah you're definitely right - i think i just mindlessly copied something from before :/